src/server/db_slice.cc

```diff
@@ -1237,6 +1237,26 @@ void DbSlice::ScheduleForOffloadStep(DbIndex db_indx, size_t increase_goal_bytes
   }
 }
 
+void DbSlice::DefragSecondTierStep(DbIndex db_indx) {
+  DCHECK(shard_owner()->tiered_storage());
+  FiberAtomicGuard guard;
+  PrimeTable& pt = db_arr_[db_indx]->prime;
+
+  static PrimeTable::Cursor cursor;
+
+  auto cb = [&](PrimeIterator it) {
+    // if it is already in the second tier
+    if (it->first.IsExternal()) {
+      shard_owner()->tiered_storage()->Defrag(db_indx, it);
+    }
+  };
+
+  // Traverse a single segment every time this function is called.
+  for (int i = 0; i < 10; ++i) {
```

Comment:
Why is this hardcoded to 10?

Reply:
yeah... this will be changed to a configurable parameter... right now this section's design may still need to be changed...